### PR TITLE
Sign up, Sign in out using Devise

### PR DESCRIPTION
**Pull request by SSalekin**

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "off" %>
```
Comment (abdullahallmehedi) [inline app/views/devise/registrations/new.html.erb]: :off

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "off" %>
```
Comment (abdullahallmehedi) [inline app/views/devise/registrations/new.html.erb]: :off

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
```
Comment (Rownaknira) [inline app/views/devise/registrations/new.html.erb]: indent

File: config/initializers/devise.rb [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` on Rails 4+ applications as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = 'e69390fa08e39478333baeed335344e3989bf1bb61ded4da0f7a379e665c3faadf0e28a0849affc22574af1c832739b91379ae62f2b67544f3551fb9edf8d0cf'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (abdullahallmehedi) [inline config/initializers/devise.rb]: ""

File: config/initializers/devise.rb [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+  # Configure the class responsible to send e-mails.
+  # config.mailer = 'Devise::Mailer'
+
+  # ==> ORM configuration
+  # Load and configure the ORM. Supports :active_record (default) and
+  # :mongoid (bson_ext recommended) by default. Other ORMs may be
+  # available as additional gems.
+  require 'devise/orm/active_record'
```
Comment (abdullahallmehedi) [inline config/initializers/devise.rb]: ""

File: config/locales/devise.en.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      unlocked: Your account has been unlocked successfully. Please sign in to continue.
+  errors:
+    messages:
+      already_confirmed: was already confirmed, please try signing in
+      confirmation_period_expired: needs to be confirmed within %{period}, please request a new one
+      expired: has expired, please request a new one
+      not_found: not found
+      not_locked: was not locked
+      not_saved:
+        one: '1 error prohibited this %{resource} from being saved:'
```
Comment (abdullahallmehedi) [inline config/locales/devise.en.yml]: ""

File: config/locales/devise.en.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        other: '%{count} errors prohibited this %{resource} from being saved:'
```
Comment (abdullahallmehedi) [inline config/locales/devise.en.yml]: ""

File: db/migrate/20151231060959_add_devise_to_users.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
+class AddDeviseToUsers < ActiveRecord::Migration
+  def self.up
+    change_table(:users) do |t|
+      ## Database authenticatable
+      t.change :email, :string,     :null => false, :default => ""
```
Comment (abdullahallmehedi) [inline db/migrate/20151231060959_add_devise_to_users.rb]: convention + indentation

File: app/views/layouts/_footer.html.erb [hunk 9]
```diff
@@ -6,8 +6,8 @@
   </div>
   <nav>
     <ul>
-      <li><%= link_to t(:about), about_path %></li>
-      <li><%= link_to t(:contact), contact_path %></li>
```
Comment (noor-framgia) [inline app/views/layouts/_footer.html.erb]: please keep it

File: app/views/layouts/_footer.html.erb [hunk 10]
```diff
@@ -6,8 +6,8 @@
   </div>
   <nav>
     <ul>
-      <li><%= link_to t(:about), about_path %></li>
```
Comment (noor-framgia) [inline app/views/layouts/_footer.html.erb]: please keep it

File: app/views/layouts/_header.html.erb [hunk 11]
```diff
@@ -1,10 +1,10 @@
 <header class="navbar navbar-fixed-top navbar-inverse">
   <div class="container">
-    <%= link_to t(:home_head), root_path, id: "logo" %>
```
Comment (noor-framgia) [inline app/views/layouts/_header.html.erb]: keep it please

File: db/migrate/20151231060959_add_devise_to_users.rb (hunk 8 above)
Comment (SSalekin) [inline db/migrate/20151231060959_add_devise_to_users.rb]: do we need conventions for migration files too?

**Comment by noor-framgia** [comment]
okey

**Comment by abdullahallmehedi** [comment]
ok

**Pull request merged**